operator/v1/types_storage.go

```diff
@@ -26,9 +26,24 @@ type Storage struct {
 	Status StorageStatus `json:"status"`
 }
 
+// CSIMigrationToggle indicates whether CSI migration should be enabled for drivers where it is optional.
+// +kubebuilder:validation:Enum=Enabled
+type CSIMigrationToggle string
+
+const (
+	CSIMigrationEnabled CSIMigrationToggle = "Enabled"
+)
+
 // StorageSpec is the specification of the desired behavior of the cluster storage operator.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.csiMigration) || has(self.csiMigration)", message="CSIMigration is required once set"
 type StorageSpec struct {
 	OperatorSpec `json:",inline"`
+
+	// CSIMigration enables CSI migration for drivers where it is optional.
+	// This field is immutable once it is set and can not be undone.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="CSIMigration is immutable"
```

Comment:
These rules came from the example [here](https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification). It should allow setting a value exactly once, but only if the field is optional in the first place. After that, the value can't change.

Reply:
From the [docs](https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/#crd-transition-rules), regarding `self == oldSelf`: "For an optional field, only allow transitioning from unset to set, or from set to unset."
And the rule for StorageSpec prevents transitioning from set to unset. I think this will work, but needs to be tested.